dir/path.go

```diff
@@ -1,64 +1,108 @@
+	LibexecFS    UnionDirFS
+	UserConfigFS UnionDirFS
+}
+
+func errorHandler(path string, err error) string {
+	// if path does not exist, return path for creating file.
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		panic(err)
+	}
+	return path
 }
 
 // Config returns the path of config.json
-func (p *PathManager) Config() (string, error) {
-	return p.ConfigFS.GetPath("config.json")
+func (p *PathManager) Config() string {
+	return errorHandler(p.ConfigFS.GetPath(ConfigFile))
 }
 
 // LocalKey returns path of the local private keys or certificate
 // in the localkeys directory
-func (p *PathManager) Localkey(keyName string) (string, error) {
-	return p.ConfigFS.GetPath("localkeys", keyName)
+func (p *PathManager) Localkey(name string, extension string) string {
+	if extension != KeyExtension && extension != CertificateExtension {
+		panic(fmt.Sprintf("doesn't support the extension `%s`", extension))
+	}
+	return errorHandler(p.UserConfigFS.GetPath(LocalKeysDir, name+extension))
 }
 
 // SigningKeyConfig return the path of signingkeys.json files
-func (p *PathManager) SigningKeyConfig() (string, error) {
-	return p.ConfigFS.GetPath("signingkeys.json")
+func (p *PathManager) SigningKeyConfig() string {
+	return errorHandler(p.UserConfigFS.GetPath(SigningKeysFile))
 }
 
 // TrustPolicy returns the path of trustpolicy.json file
-func (p *PathManager) TrustPolicy() (string, error) {
-	return p.ConfigFS.GetPath("trustpolicy.json")
+func (p *PathManager) TrustPolicy() string {
+	return errorHandler(p.ConfigFS.GetPath(TrustPolicyFile))
 }
 
 // X509TrustStore returns the path of x509 trust store certificate
-func (p *PathManager) X509TrustStore(prefix, namedStore string) (string, error) {
-	return p.ConfigFS.GetPath("truststore", "x509", prefix, namedStore)
+func (p *PathManager) X509TrustStore(prefix, namedStore string) string {
+	return errorHandler(p.ConfigFS.GetPath(TrustStoreDir, "x509", prefix, namedStore))
 }
 
 // CachedSignature returns the cached signature file path
-func (p *PathManager) CachedSignature(manifestDigest, signatureDigest digest.Digest) (string, error) {
-	return p.CacheFS.GetPath(
-		"signatures",
+func (p *PathManager) CachedSignature(manifestDigest, signatureDigest digest.Digest) string {
+	return errorHandler(p.CacheFS.GetPath(
+		SignatureStoreDirName,
 		manifestDigest.Algorithm().String(),
 		manifestDigest.Encoded(),
 		signatureDigest.Algorithm().String(),
 		signatureDigest.Encoded()+SignatureExtension,
-	)
+	))
 }
 
 // CachedSignatureRoot returns the cached signature root path
-func (p *PathManager) CachedSignatureRoot(manifestDigest digest.Digest) (string, error) {
-	return p.CacheFS.GetPath(
-		"signatures",
+func (p *PathManager) CachedSignatureRoot(manifestDigest digest.Digest) string {
+	return errorHandler(p.CacheFS.GetPath(
+		SignatureStoreDirName,
 		manifestDigest.Algorithm().String(),
 		manifestDigest.Encoded(),
-	)
+	))
+}
+
+func (p *PathManager) CachedSignatureStoreDirPath() string {
```

Comment:
Documentation requested for this method.

Reply:
Done.